packages/assets-controllers/src/NftDetectionController.ts

```diff
@@ -588,6 +610,80 @@ export class NftDetectionController extends BaseController<
               ? elm.blockaidResult?.result_type === BlockaidResultType.Benign
               : true),
         );
+        // Retrieve collections from apiNfts
+        // contract and collection.id are equal for simple contract addresses; this is to exclude cases for shared contracts
+        const collections = apiNfts.reduce<string[]>((acc, currValue) => {
+          if (
+            !acc.includes(currValue.token.contract) &&
+            currValue.token.contract === currValue?.token?.collection?.id
+          ) {
+            acc.push(currValue.token.contract);
+          }
+          return acc;
+        }, []);
+
+        if (collections.length !== 0) {
+          // Call API to retrive collections infos
+          // The api accept a max of 20 contracts
+          const collectionResponse: GetCollectionsResponse =
+            await reduceInBatchesSerially({
+              values: collections,
+              batchSize: MAX_GET_COLLECTION_BATCH_SIZE,
+              eachBatch: async (allResponses, batch) => {
+                const params = new URLSearchParams(
+                  batch.map((s) => ['contract', s]),
+                );
+                params.append('chainId', '1'); // Adding chainId 1 because we are only detecting for mainnet
+                const collectionResponseForBatch = await fetchWithErrorHandling(
+                  {
+                    url: `${
+                      NFT_API_BASE_URL as string
+                    }/collections?${params.toString()}`,
+                    options: {
+                      headers: {
+                        Version: NFT_API_VERSION,
+                      },
+                    },
+                    timeout: NFT_API_TIMEOUT,
+                  },
+                );
+
+                return {
+                  ...allResponses,
+                  ...collectionResponseForBatch,
+                };
+              },
+              initialResult: {},
+            });
+
+          // Add collections response fields to  newnfts
+          if (collectionResponse.collections?.length) {
+            apiNfts.forEach((singleNFT) => {
+              const found = collectionResponse.collections.find(
+                (elm) =>
+                  elm.id?.toLowerCase() ===
+                  singleNFT.token.contract.toLowerCase(),
+              );
+              if (found) {
+                singleNFT.token = {
+                  ...singleNFT.token,
+                  collection: {
+                    ...(singleNFT.token.collection
+                      ? singleNFT.token.collection
+                      : {}),
```

Comment:
Nit:

``` suggestion
                    ...(singleNFT.token.collection ?? {}),

Reply:
I think both would achieve the same result, will update to your suggestion that is more concise 🙏